modules/Movecraft/src/main/java/net/countercraft/movecraft/craft/ChunkManager.java

```diff
@@ -0,0 +1,52 @@
+package net.countercraft.movecraft.craft;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.bukkit.Chunk;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.world.ChunkUnloadEvent;
+import org.bukkit.scheduler.BukkitRunnable;
+
+import net.countercraft.movecraft.Movecraft;
+
+public class ChunkManager implements Listener {
+	
+	private static List<Chunk> allChunks = new ArrayList<Chunk>();
```

Comment:
prefer chunks over allChunks

Reply:
Also, you should use a either weak reference to the chunks, or store by chunk cordinates. You could run into some issues with world saving and server shutdown, as well as plugin reloads (which we generally don't support but we don't want to break stuff if unnecessary)